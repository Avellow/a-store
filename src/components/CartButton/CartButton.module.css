.button[data-animation="1"] {
  animation: scale 0.7s;
}

@keyframes scale {
  50% {
    transform: scale(1.2);
  }
  100% {
    transform: scale(1);
  }
}
