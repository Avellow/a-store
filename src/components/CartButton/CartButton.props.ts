import { IconButtonProps } from "@alfalab/core-components/icon-button";

export interface CartButtonProps extends Omit<IconButtonProps, "icon"> {
  goodsQuantity: number;
}
