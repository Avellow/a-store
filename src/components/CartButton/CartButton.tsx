import { IconButton } from '@alfalab/core-components/icon-button';
import { Circle } from '@alfalab/core-components/icon-view/circle';
import { ShoppingBagMIcon } from '@alfalab/icons-glyph/ShoppingBagMIcon';
import { Badge } from '@alfalab/core-components/badge';
import { useEffect, useState } from 'react';
import cn from 'classnames';

import styles from './CartButton.module.css';
import { redColor, whiteColor } from '../../vendor/constants';
import { CartButtonProps } from './CartButton.props';

let lastQuantity = 0;

export const CartButton = ({ goodsQuantity, className, ...restProps }: CartButtonProps): JSX.Element => {
  const CartIconView = () => (
    <Circle
      size={80}
      backgroundColor={redColor}
      bottomAddons={<Badge view='count' content={goodsQuantity} />}
      dataTestId='cart-size'
    >
      <ShoppingBagMIcon fill={redColor} stroke={whiteColor} />
    </Circle>
  );

  const [animation, setAnimation] = useState<0 | 1>(0);

  useEffect(() => {
    if (lastQuantity !== goodsQuantity) {
      lastQuantity = goodsQuantity;
      setAnimation(1);
    }
  }, [goodsQuantity]);

  return (
    <IconButton
      key={goodsQuantity}

      className={cn(className, styles.button)}
      icon={CartIconView}
      {...restProps}

      data-animation={animation}
    />
  );
};
