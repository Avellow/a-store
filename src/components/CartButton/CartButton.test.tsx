import { render, screen, configure } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { CartButton } from './CartButton';

configure({ testIdAttribute: 'data-test-id' });

describe('Cart button component view', () => {

  it('should be rendered', () => {
    expect(render(<CartButton goodsQuantity={11} />)).toMatchSnapshot();
  });

  it('should show cart-size correctly', () => {
    render(<CartButton goodsQuantity={11} />);
    expect(screen.getByTestId('cart-size')).toHaveTextContent('11')
  });

  it('should show `99+` if cart-size more than 99', () => {
    render(<CartButton goodsQuantity={1000} />);
    expect(screen.getByTestId('cart-size')).toHaveTextContent('99+');
  });
});
