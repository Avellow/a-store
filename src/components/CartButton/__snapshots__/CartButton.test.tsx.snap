// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart button component view should be rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button button icon-button__primary_7jncp icon-button__component_7jncp"
        type="button"
      >
        <span
          class="button__text_onnkc button__stretchText_onnkc"
        >
          <span
            class="icon-button__iconWrapper_gy47p icon-button__s_gy47p"
          >
            <div
              class="icon-view__componentWrapper_1wtym icon-view__size_80_1wtym"
              data-test-id="cart-size"
            >
              <div
                class="icon-view__component_1wtym"
              >
                <svg
                  focusable="false"
                  height="80"
                  viewBox="0 0 80 80"
                  width="80"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="icon-view__bg_1wtym"
                    d="M71 56C74.0263 56 77.2805 54.8407 78.1837 51.9523C79.364 48.178 80 44.1633 80 40C80 17.9086 62.0914 0 40 0C17.9086 0 0 17.9086 0 40C0 62.0914 17.9086 80 40 80C44.1633 80 48.178 79.364 51.9523 78.1837C54.8407 77.2805 56 74.0263 56 71C56 62.7157 62.7157 56 71 56Z"
                    style="fill: rgb(239, 49, 36);"
                  />
                </svg>
                <div
                  class="icon-view__children_1wtym"
                >
                  <svg
                    fill="rgb(239, 49, 36)"
                    focusable="false"
                    height="24"
                    role="img"
                    stroke="rgb(255, 255, 255)"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16 5v1h4l1 16H3L4 6h4V5a4 4 0 118 0zm-4-2a2 2 0 012 2v1h-4V5a2 2 0 012-2zm-2 6a1 1 0 11-2 0 1 1 0 012 0zm6 0a1 1 0 11-2 0 1 1 0 012 0z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="icon-view__addons_1wtym icon-view__bottomAddons_1wtym"
              >
                <div
                  class="badge__count_1vi3d"
                >
                  <div
                    class="badge__component_1vi3d badge__m_1vi3d badge__count_1vi3d badge__heightS_1vi3d"
                    style="height: 16px; min-width: 16px;"
                  >
                    11
                  </div>
                </div>
              </div>
            </div>
          </span>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button button icon-button__primary_7jncp icon-button__component_7jncp"
      type="button"
    >
      <span
        class="button__text_onnkc button__stretchText_onnkc"
      >
        <span
          class="icon-button__iconWrapper_gy47p icon-button__s_gy47p"
        >
          <div
            class="icon-view__componentWrapper_1wtym icon-view__size_80_1wtym"
            data-test-id="cart-size"
          >
            <div
              class="icon-view__component_1wtym"
            >
              <svg
                focusable="false"
                height="80"
                viewBox="0 0 80 80"
                width="80"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="icon-view__bg_1wtym"
                  d="M71 56C74.0263 56 77.2805 54.8407 78.1837 51.9523C79.364 48.178 80 44.1633 80 40C80 17.9086 62.0914 0 40 0C17.9086 0 0 17.9086 0 40C0 62.0914 17.9086 80 40 80C44.1633 80 48.178 79.364 51.9523 78.1837C54.8407 77.2805 56 74.0263 56 71C56 62.7157 62.7157 56 71 56Z"
                  style="fill: rgb(239, 49, 36);"
                />
              </svg>
              <div
                class="icon-view__children_1wtym"
              >
                <svg
                  fill="rgb(239, 49, 36)"
                  focusable="false"
                  height="24"
                  role="img"
                  stroke="rgb(255, 255, 255)"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M16 5v1h4l1 16H3L4 6h4V5a4 4 0 118 0zm-4-2a2 2 0 012 2v1h-4V5a2 2 0 012-2zm-2 6a1 1 0 11-2 0 1 1 0 012 0zm6 0a1 1 0 11-2 0 1 1 0 012 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <div
              class="icon-view__addons_1wtym icon-view__bottomAddons_1wtym"
            >
              <div
                class="badge__count_1vi3d"
              >
                <div
                  class="badge__component_1vi3d badge__m_1vi3d badge__count_1vi3d badge__heightS_1vi3d"
                  style="height: 16px; min-width: 16px;"
                >
                  11
                </div>
              </div>
            </div>
          </div>
        </span>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
